src/crimson/os/seastore/cached_extent.h

```diff
@@ -543,54 +547,73 @@ class CachedExtent
 
   /// Returns true if extent is stable, written and shared among transactions
   bool is_stable_written() const {
-    return state == extent_state_t::CLEAN_PENDING ||
-      state == extent_state_t::CLEAN ||
-      state == extent_state_t::DIRTY;
+    return state == extent_state_t::CLEAN
+           || state == extent_state_t::DIRTY;
   }
 
   bool is_stable_writting() const {
-    // MUTATION_PENDING and under-io extents are already stable and visible,
-    // see prepare_record().
+    // mutated/INITIAL_PENDING and under-io extents are already
+    // stable and visible, see prepare_record().
     //
-    // XXX: It might be good to mark this case as DIRTY from the definition,
+    // XXX: It might be good to mark this case as DIRTY/CLEAN from the definition,
     // which probably can make things simpler.
-    return is_mutation_pending() && is_pending_io();
+    if (is_pending_io()) {
+      assert(has_mutation() || is_initial_pending());
```

Comment:
This fails UT and it borrows from https://github.com/ceph/ceph/pull/63218/commits/d91be79a9bf7e1a86a1969efe8a70c7db1fea927, https://github.com/ceph/ceph/pull/63218, will reply there.

Reply:
modified inplace as suggested.